### ISSUE TITLE
Support gometalinter 2.0

### ISSUE THREAD

**Issue by derekperkins** [Microsoft/vscode-go#1393]
"The configuration format has also changed, but should be mostly backwards compatible.

Another major change is that from now on we will be releasing binary packages. gometalinter's management of linter installation will soon be deprecated, and eventually removed."

https://github.com/alecthomas/gometalinter/releases/tag/v2.0.0

**Comment by ramya-rao-a** [comment]
PRs are welcome

**Comment by ramya-rao-a** [comment]
I am confused a bit here. What changes are needed from the Go extension's side to use gometalinter 2.0?
Doesnt it suffice that the user updates their version locally?

cc'ing everyone who upvoted this feature request: @atishpatel, @zmb3, @taketori, @SNieroda, @fishhead108

**Comment by zerkms** [comment]
@ramya-rao-a in the config gometalinter v2 is underlined as a wrong value

**Comment by ramya-rao-a** [comment]
@zerkms You mean setting `go.lintTool` to `gometalinter.v2`? Doesn't running a `go get -u github.com/alecthomas/gometalinter` update the tool to use v2 ?

**Comment by zerkms** [comment]
@ramya-rao-a that's right - if you set `go.lintTool` to `gometalinter.v2` underlines it as not a valid value.

It still works, but in the config it is marked as an invalid value.

**Comment by ramya-rao-a** [comment]
@zmb3 I just did a `go get` on gometalinter and was able to get the latest and the binary is built. This surely has all the new features.

cc @alecthomas for clarification

**Comment by zerkms** [comment]
@ramya-rao-a if I set it to `"go.lintTool": "gometalinter",` no validation happens at all.

```
➜ which gometalinter
gometalinter not found
```
I'm not sure how it's supposed to work

**Comment by ramya-rao-a** [comment]
@zerkms Please run `go get -u gopkg.in/alecthomas/gometalinter` and try again

**Comment by zerkms** [comment]
@ramya-rao-a as it is stated on the `gometalinter` readme it would install the unstable version. I'm sure it would work, I'm not sure I'm ready to run unstable version of devtools though.

**Comment by ramya-rao-a** [comment]
@zerkms Every [tool that this Go extension](https://github.com/Microsoft/vscode-go/wiki/Go-tools-that-the-Go-extension-depends-on) uses comes from the master branch of the corresponding repo. We dont use the one from `gopkg.in` for the exact issue that we then have to support all the versions as the binary name would be different.

https://github.com/Microsoft/vscode-go/pull/1297 is tracking a feature where you can map a tool used by this extension to the path where it needs to find the tool. Once that feature is in, you can add a mapping like `"gometalinter": "gometalinter.v2"` and that should help your case.

**Comment by zerkms** [comment]
I see thanks for the detailed response, now I see. Thanks a lot.
